iceoryx_hoofs/include/iceoryx_hoofs/cxx/vector.hpp

```diff
@@ -220,21 +221,29 @@ class vector
 
     void clearFrom(const uint64_t startPosition) noexcept;
 
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : safe access is guaranteed since the C-style array is wrapped
+    // inside the vector class
     /// @todo iox-#1614 Replace with UninitializedArray
     // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays,hicpp-avoid-c-arrays)
     using element_t = uint8_t[sizeof(T)];
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : safe access is guaranteed since the C-style array is wrapped
+    // inside the vector class
     alignas(T) element_t m_data[Capacity];
     // NOLINTEND(cppcoreguidelines-avoid-c-arrays,hicpp-avoid-c-arrays)
     uint64_t m_size{0U};
 };
-} // namespace cxx
-} // namespace iox
 
+// AXIVION Next Construct AutosarC++19_03-A13.5.5 : intentional implementation with different parameters to enable
+// comparison of vectors with different capacity
 template <typename T, uint64_t CapacityLeft, uint64_t CapacityRight>
-bool operator==(const iox::cxx::vector<T, CapacityLeft>& lhs, const iox::cxx::vector<T, CapacityRight>& rhs) noexcept;
+bool operator==(const vector<T, CapacityLeft>& lhs, const vector<T, CapacityRight>& rhs) noexcept;
```

Comment:
Those operators could also be constexpr. If you have to adjust something in this PR you could add it ... otherwise you can do this when you visit the vector the next time.